mlir/utils/jenkins/Dockerfile

```diff
@@ -12,66 +12,79 @@ RUN apt-get clean all
 
 # --------------------- Section 1: ported from llvm-premerge-checks -----------------
 # Keep this section up-to-date with the upstream
-# https://github.com/google/llvm-premerge-checks/blob/master/containers/base-debian/Dockerfile
```

Comment:
Most of these diffs are just changes in the upstream source.  I did remove installation of gosu, tini, autoconf, automake, and libtool that I don't think we use.

Reply:
Also, should we move to UBuntu 22.04 here?